### PR TITLE
Visitor can select which genres to filter by

### PR DESCRIPTION
**Pull request by kermit-klein**
[PT LINK 1](https://www.pivotaltracker.com/story/show/172766702)
[PT LINK 2](https://www.pivotaltracker.com/story/show/172770320)

```
as a user
in order to filter my searches
I would like an option to select genres
```
```
As a visitor
In order to get a sense of what the app can provide me
I want to be able to track one person without having an account
I want to see a list of upcoming releases featuring that person
```
**added functions**
button toggle color control
genres select function to add/remove genres to App.jsx state
get request with genre selections and actor from search to get upcoming release list
back button on release upcoming list

**TDD**
visitor can toggle on/off genres selections
visitor can select one person to see track results

**package added**
semantic ui

PS:Pauline helped and had commits yesterday's spike branch

### CODE REVIEW

**Review by PaulineBA — APPROVED** [review]

**Review by erikbjoern — APPROVED** [review]
We could maybe think of ways to refactor the way we select/deselect genres, but as long it works it's good! Nice work!

**Comment by kermit-klein** [comment]
> We could maybe think of ways to refactor the way we select/deselect genres, but as long it works it's good! Nice work!

"as long it works" THE KEY PHRASE HERE

File: cypress/integration/userCanTrackOnePerson.feature.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        cy.contains("2023-11-02");
+        cy.contains("Actor");
+        cy.contains("stuff happened");
+        cy.get("img").should("be.visible");
+      });
+    });
+
+    it("User can go back to previous page", () => {
+      cy.get("#btn-back").click();
+      cy.get("#header").should("contain", "Release Tracker");
```
Comment (erikbjoern) [inline cypress/integration/userCanTrackOnePerson.feature.js]: I could argue the header should be visible in every view/page, as in the Lo-Fi:s

File: src/App.jsx [hunk 2]
```diff
@@ -21,15 +28,67 @@ export class App extends Component {
...
+      });
+    } else {
+      this.setState({
+        genresSelected: [...this.state.genresSelected, selectedGenre],
+      });
+    }
+  };
+
+  resetMoviePerson = () => {
+    this.setState({ moviePersonResult: [] });
```
Comment (erikbjoern) [inline src/App.jsx]: should this reset activeName as well?

File: src/components/Genres.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        </Button>
+        <Button
+          toggle
+          onClick={(e) => {
+            this.props.genresHandler(e);
+            this.btnActivity(e);
+          }}
+          id="thriller"
+          name="thriller"
+          className={this.state.thriller ? "active" : ""}
```
Comment (erikbjoern) [inline src/components/Genres.jsx]: Should it say className or active? They buttons are different here.

**Review by erikbjoern — COMMENTED** [review]
Left a few comments

File: src/App.jsx [hunk 4]
```diff
@@ -21,15 +28,66 @@ export class App extends Component {
...
+    e.preventDefault();
+    let id = e.target.dataset.id;
+    this.setState({ activeName: e.target.text });
+    try {
+      const response = await axios.get(`/api/v1/movie_person/${id}`, {
+        params: { genres: this.state.genresSelected },
+      });
+      this.setState({ moviePersonResult: response.data.result.movies });
+    } catch (error) {
+      debugger;
```
Comment (jysmys) [inline src/App.jsx]: remove all debugger

**Review by jysmys — APPROVED** [review]

**Pull request merged**